internal/stage_f2.go

```diff
@@ -2,73 +2,94 @@ package internal
-	"github.com/codecrafters-io/kafka-tester/protocol/legacy_kafkaapi"
-	"github.com/codecrafters-io/kafka-tester/protocol/legacy_serializer"
+	"github.com/codecrafters-io/kafka-tester/internal/response_asserter"
+	"github.com/codecrafters-io/kafka-tester/internal/response_assertions"
+	"github.com/codecrafters-io/kafka-tester/internal/response_decoders"
+	"github.com/codecrafters-io/kafka-tester/protocol/builder"
+	"github.com/codecrafters-io/kafka-tester/protocol/kafka_client"
+	"github.com/codecrafters-io/kafka-tester/protocol/kafka_files_generator"
+	"github.com/codecrafters-io/kafka-tester/protocol/kafkaapi"
+	"github.com/codecrafters-io/tester-utils/logger"
 	"github.com/codecrafters-io/tester-utils/test_case_harness"
 )
 
 func testFetchWithNoTopics(stageHarness *test_case_harness.TestCaseHarness) error {
-	b := kafka_executable.NewKafkaExecutable(stageHarness)
 	stageLogger := stageHarness.Logger
-	err := legacy_serializer.GenerateLogDirs(stageLogger, false)
-	if err != nil {
+
+	if err := kafka_files_generator.NewFilesHandler(logger.GetQuietLogger("")).GenerateServerConfiguration(); err != nil {
 		return err
 	}
 
+	b := kafka_executable.NewKafkaExecutable(stageHarness)
+
 	if err := b.Run(); err != nil {
 		return err
 	}
 
-	correlationId := getRandomCorrelationId()
+	client := kafka_client.NewClient("localhost:9092")
 
-	client := legacy_kafka_client.NewClient("localhost:9092")
 	if err := client.ConnectWithRetries(b, stageLogger); err != nil {
 		return err
 	}
-	defer func(client *legacy_kafka_client.Client) {
-		_ = client.Close()
-	}(client)
-
-	request := legacy_kafkaapi.FetchRequest{
-		Header: legacy_builder.NewRequestHeaderBuilder().BuildFetchRequestHeader(correlationId),
-		Body: legacy_kafkaapi.FetchRequestBody{
-			MaxWaitMS:         500,
-			MinBytes:          1,
-			MaxBytes:          52428800,
-			IsolationLevel:    0,
-			FetchSessionID:    0,
-			FetchSessionEpoch: 0,
-			Topics:            []legacy_kafkaapi.Topic{},
-			ForgottenTopics:   []legacy_kafkaapi.ForgottenTopic{},
-			RackID:            "",
-		},
-	}
 
-	response, err := client.SendAndReceive(request, stageLogger)
-	if err != nil {
-		return err
-	}
+	defer client.Close()
 
-	responseHeader, responseBody, err := legacy_kafkaapi.DecodeFetchHeaderAndResponse(response.Payload, 16, stageLogger)
+	correlationId := getRandomCorrelationId()
+
+	request := builder.NewFetchRequstWithEmptyTopicsBuilder().
+		WithCorrelationId(correlationId).
+		Build()
+
+	rawResponse, err := client.SendAndReceive(request, stageLogger)
 	if err != nil {
 		return err
 	}
 
-	expectedResponseHeader := legacy_builder.BuildResponseHeader(correlationId)
-	if err = legacy_assertions.NewResponseHeaderAssertion(*responseHeader, expectedResponseHeader).Run(stageLogger); err != nil {
-		return err
-	}
+	assertion := response_assertions.NewFetchResponseAssertion().
+		ExpectCorrelationId(correlationId).
+		ExpectErrorCodeInBody(0).
+		ExpectThrottleTimeMs(0)
```

Comment:
Oh, didn't realize that we assert `ThrottleTimeMs`. Can't we do this for other response assertions too then?